src/modules/personenkontext/domain/personenkontexte-update.ts

```diff
@@ -243,13 +244,72 @@ export class PersonenkontexteUpdate {
         return createdPKs;
     }
 
+    private async validationForRollenartLern(
+        existingPKs: Personenkontext<true>[],
+        sentPKs: Personenkontext<boolean>[],
+    ): Promise<Option<PersonenkontexteUpdateError>> {
+        // Check if the sent PKs have any LERN roles
+        const sentRollen: Rolle<true>[] = await this.getUniqueRollenFromPersonenkontexte(sentPKs);
+        const hasAnyLernInSent: boolean = sentRollen.some((rolle: Rolle<true>) => rolle.rollenart === RollenArt.LERN);
+
+        // Check if existing PKs have LERN roles
+        const hasLernInExisting: boolean = await this.hasAnyPersonenkontextWithRollenartLern(existingPKs);
+
+        // Early return: If neither the existing PKs nor the sent PKs contain LERN roles, it's safe to return undefined
+        if (!hasLernInExisting && !hasAnyLernInSent) {
+            return undefined;
+        }
```

Comment:
Ah, I see my concerns from the previous comment are addressed here. But now we're maintaining two seperate pieces of code. Is there no way to use the specification for both? (Maybe instead of giving it a single Kontext it could accept an array of kontexte)

Reply:
Me and Phael thought it was better to keep the update aggregate isolated and clean since these two are different use cases and so each should have it's own logic. On top of that, one Endpoint can only send 1 PK, the other a bunch of them. But from another POV you could see it as a redundant check because it can be solved by allowing an array of kontexte (but that would require again a FE change and maybe could affect migration since it's used there)

Reply:
You could keep the FE the same. In `src/modules/personenkontext/specification/personenkontext-klasse-specification.ts` you could just write something like this (Use `[p]` as the parameter instead of `p`)
```ts
if (!(await this.nurRollenartLern.checkRollenartLern([p]))) {
    return new UpdateInvalidRollenartForLernError();
}
```

Then both the workflow aggregate and the "old" service could use the specification